File: src/test/java/com/github/k0zka/contentcompress/ContentGzipMojoTest.java
package com.github.k0zka.contentcompress;

import java.io.File;
import java.io.IOException;

import org.apache.commons.io.FileDeleteStrategy;
import org.apache.commons.io.FileUtils;
import org.junit.Before;
import org.junit.Test;

import com.github.k0zka.contentcompress.ContentGzipMojo;

import static junit.framework.TestCase.assertTrue;
import static org.junit.Assert.assertFalse;

public class ContentGzipMojoTest {

	File basedir;
	
	@Before
	public void createWebappDirectoryStructure() throws IOException {
		basedir = new File("target/testdir");
		if(basedir.exists()) {
			FileDeleteStrategy.FORCE.delete(basedir);
		}
		basedir.mkdirs();
		FileUtils.copyDirectory(new File("src/test/resources"), basedir);
	}

	@Test
	public void testSeekAndGzip() throws IOException {
		new ContentGzipMojo().seekAndGzip(basedir);
		assertTrue(new File(basedir, "skins/default.css.gz").exists());
		assertTrue(new File(basedir, "index.html.gz").exists());
	}

	@Test
	public void testSeekAndGzipNonRecursive() throws IOException {
		ContentGzipMojo gzipMojo = new ContentGzipMojo();
		gzipMojo.recursive = false;
		gzipMojo.seekAndGzip(basedir);
		assertFalse(new File(basedir, "skins/default.css.gz").exists());
		assertTrue(new File(basedir, "index.html.gz").exists());
	}

}

File: src/test/kotlin/com/github/k0zka/contentcompress/ContentGzipMojoTest.kt
package com.github.k0zka.contentcompress

import java.io.File
import java.io.IOException

import org.apache.commons.io.FileDeleteStrategy
import org.apache.commons.io.FileUtils
import org.junit.Before
import org.junit.Test

import com.github.k0zka.contentcompress.ContentGzipMojo

import junit.framework.TestCase.assertTrue
import org.junit.Assert.assertFalse

class ContentGzipMojoTest {

	internal lateinit var basedir: File

	@Before
	@Throws(IOException::class)
	fun createWebappDirectoryStructure() {
		basedir = File("target/testdir")
		if (basedir.exists()) {
			FileDeleteStrategy.FORCE.delete(basedir)
		}
		basedir.mkdirs()
		FileUtils.copyDirectory(File("src/test/resources"), basedir)
	}

	@Test
	@Throws(IOException::class)
	fun testSeekAndGzip() {
		ContentGzipMojo().seekAndGzip(basedir)
		assertTrue(File(basedir, "skins/default.css.gz").exists())
		assertTrue(File(basedir, "index.html.gz").exists())
	}

	@Test
	@Throws(IOException::class)
	fun testSeekAndGzipNonRecursive() {
		val gzipMojo = ContentGzipMojo()
		gzipMojo.recursive = false
		gzipMojo.seekAndGzip(basedir)
		assertFalse(File(basedir, "skins/default.css.gz").exists())
		assertTrue(File(basedir, "index.html.gz").exists())
	}

}
